client/rest-high-level/src/test/java/org/elasticsearch/client/tasks/CancelTasksResponseTests.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.tasks;
+
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class CancelTasksResponseTests extends ESTestCase {
```

Comment:
a clarification here:

`AbstractResponseTestCase<CancelTasksResponse, CancelTasksResponseTests.CancelTasksResponse> `

the first `CancelTasksResponse` is the server side object (that also extends `toXContent`.
Shall I use this to create an instance that gets marshalled in XCOntent and parsed back?
I assumed you wanted to remove dependencies from the server, but maybe for testing purposes it is useful and more realistic to test the marshalling and unmarshalling of data.

If so, instead of using the `CancelTasksResponseTests.CancelTasksResponse` shall I use the real tasks counterpart, that implements all the `fromXContent` already (`org.elasticsearch.client.tasks.CancelTasksResponse`) ?
Otherwise there are duplications of the logic.

Reply:
also, shall I remove my test or can I Keep it renaming it?

Reply:
The test is not needed as is, but once it extends the `AbstractResponseTestCase` it will be good. 

> I assumed you wanted to remove dependencies from the server, but maybe for testing purposes it is useful and more realistic to test the marshalling and unmarshalling of data.

This is correct, we want to test the actual server object transforming to a client. There are a lot of tests that extend this so you can see the pattern for usage here.

> If so, instead of using the CancelTasksResponseTests.CancelTasksResponse shall I use the real tasks counterpart, 

Yes, the CancelTaskResponseTests subclass was only to show you how to use it, so it can be viewed as an example, and not code meant to be in the pull request.

Reply:
ok I followed your boilerplate: actually I needed to create an adapter around the server side `CancelTaskResponse` because the default `toXContent` marshalls tasks in a different way, not under `nodes` but under `tasks`